src/org/traccar/api/resource/DeviceResource.java

```diff
@@ -122,4 +121,34 @@ public Response updateTotalDistance(DeviceTotalDistance entity) throws SQLExcept
         return Response.noContent().build();
     }
 
+    @Path("/{slave : (geofences|drivers|attributes)}")
```

Comment:
I think it's OK because it's a major refactoring. It will be a really small change for anyone who uses API directly. Just need to remind people about it with release notes.